### ISSUE TITLE
Passing `userParams` to `callbackNewPoint`

### ISSUE THREAD

**Issue by ferrolho** [JuliaOpt/KNITRO.jl#141]
The example for adding a `callbackNewPoint`
https://github.com/JuliaOpt/KNITRO.jl/blob/2ccc7a4c6b6212cb79f4d16b061a042beab55341/examples/nlp2.jl#L201-L204
seems to already use `userParams` for passing problem information to the callback:
https://github.com/JuliaOpt/KNITRO.jl/blob/2ccc7a4c6b6212cb79f4d16b061a042beab55341/examples/nlp2.jl#L111-L115

Is it possible to pass additional data by calling `KN_set_cb_user_params` as one would do for other kind of callbacks, e.g., `KN_add_eval_callback`? As in:
```julia
KNITRO.KN_set_cb_user_params(kc, cb, (data_struct_1, data_struct_2))
```

**Comment by frapac** [comment]
Indeed, the userparams in `callbackNewPoint` is mispecified. It should return a valid userparams rather than a Knitro model. I will work on a fix then.

**Comment by frapac** [comment]
Closed with #143
